lib/internal/assert/assertion_error.js

```diff
@@ -437,6 +437,23 @@ class AssertionError extends Error {
     this.name = 'AssertionError';
   }
 
+  get details() {
+    return this;
+  }
+
+  set details(details) {
+    delete this.actual;
+    delete this.expected;
+    delete this.operator;
```

Comment:
```suggestion
    this.actual = undefined;
    this.expected = undefined;
    this.operator = undefined;
```

We generally avoid using `delete` because it introduces a perf issue. Setting to `undefined` does what we need here.